sympy/combinatorics/fp_groups.py

```diff
@@ -1499,45 +1499,40 @@ def elimination_technique_1(C):
     rels = C._reidemeister_relators
     # the shorter relators are examined first so that generators selected for
     # elimination will have shorter strings as equivalent
-    rels.sort(reverse=True)
+    rels.sort()
     gens = C._schreier_generators
     redundant_gens = {}
-    contained_gens = []
+    redundant_rels = []
+    used_gens = set()
     # examine each relator in relator list for any generator occuring exactly
     # once
-    next_i = False
-    for i in range(len(rels) -1, -1, -1):
-        rel = rels[i]
-        # don't look for a new generator occuring once in relator which
-        # has already found to posses a
-        for gen in redundant_gens:
-            gen_sym = gen.array_form[0][0]
-            if any([gen_sym == r[0] for r in rel.array_form]):
-                next_i = True
-                break
-        if next_i:
-            next_i = False
+    for rel in rels:
+        # don't look for a redundant generator in a relator which
+        # depends on previously found ones
+        if any([g in rel.contains_generators() for g in redundant_gens]):
```

Comment:
`rel.contains_generators()` could be saved in a local variable to avoid repeated recomputations.